src/transport/mqtt/config/transport.rs

```diff
@@ -20,41 +20,68 @@ use std::{path::PathBuf, sync::Arc};
 
 use rumqttc::Transport;
 use rustls::pki_types::PrivatePkcs8KeyDer;
+use rustls::RootCertStore;
 use tokio::fs;
-use tracing::{debug, error, info};
+use tracing::{debug, error, info, instrument, warn};
 use url::Url;
 
+use super::{
+    tls::{ClientAuth, NoVerifier},
+    CertificateFile, PrivateKeyFile,
+};
+use crate::transport::mqtt::config::tls::read_root_cert_store;
 use crate::{
     error::Report,
     transport::mqtt::{crypto::Bundle, pairing::ApiClient, PairingError},
 };
 
-use super::{tls::ClientAuth, CertificateFile, PrivateKeyFile};
-
 /// Structure to create an authenticated [`Transport`]
 #[derive(Debug)]
 pub(crate) struct TransportProvider {
     pairing_url: Url,
     credential_secret: String,
     store_dir: Option<PathBuf>,
     insecure_ssl: bool,
+    root_cert_store: Arc<RootCertStore>,
 }
 
 impl TransportProvider {
+    #[allow(dead_code)]
     pub(crate) fn new(
         pairing_url: Url,
         credential_secret: String,
         store_dir: Option<PathBuf>,
         insecure_ssl: bool,
+        root_cert_store: RootCertStore,
     ) -> Self {
         Self {
             pairing_url,
             credential_secret,
             store_dir,
             insecure_ssl,
+            root_cert_store: Arc::new(root_cert_store),
         }
     }
```

Comment:
```suggestion
    pub(crate) fn new(
        pairing_url: Url,
        credential_secret: String,
        store_dir: Option<PathBuf>,
        insecure_ssl: bool,
        root_cert_store: RootCertStore,
    ) -> Self {
        Self {
            pairing_url,
            credential_secret,
            store_dir,
            insecure_ssl,
            root_cert_store: Arc::new(root_cert_store),
        }
    }
```

We can remove it if not used anymore